### DISCUSSION TITLE
How to define and reference an array of enums in graphql.ts?

### DISCUSSION THREAD

**Discussion by TheoMer** [prisma/prisma#3157] [Q&A] [answered]
So I have the following model structure:

```prisma
model User {
  permissions2     Permission2[]
}

enum Permission2 {
  ADMIN
  GUEST_USER
  ITEMCREATE
  ITEMDELETE
  ITEMUPDATE
  PERMISSIONUPDATE
  USER
}
```

which I wish to correctly define in my graphql.ts file:

```ts
schema.objectType({
    name: 'User',
    definition(t) {
      t.model.permissions2({ ????
        filtering: true,
        ordering: true,
      })
    },
  })

  schema.enumType({
    name: 'Permission2',
    ????
  })
```

**Comment by TheoMer** [comment]
So far what I've arrived at is:

```ts
schema.objectType({
    name: 'User',
    definition(t) {
      t.model.permissions2
    },
  })

  schema.enumType({
    name: 'Permission2',
    members: [ "ADMIN", "USER", "GUEST_USER", "ITEMCREATE", "ITEMUPDATE", "ITEMDELETE", "PERMISSIONUPDATE"]
  })
```

which generates no errors but I just need clarification that it satisfies the model.

**Comment by ryands17** [comment, marked as answer]
Hey @TheoMer 👋 
You will be able to define the enums currently like this, but filtering an array of enums isn't currently supported by Prisma and there is an issue [here](https://github.com/prisma/prisma-client-js/issues/341) that you can track at the moment. Filtering can be performed on a single enum using the normal operators.

**Comment by TheoMer** [comment]
Hey @ryands17 

Many thanks for the heads up. I've got as far as:

```ts
schema.objectType({
    name: 'User',
    definition(t) {
      t.model.permissions2() //Awaiting filtering support for arrays of enums
    },
  })

  schema.enumType({
    name: 'Permission2',
    members: [ "ADMIN", "USER", "GUEST_USER", "ITEMCREATE", "ITEMUPDATE", "ITEMDELETE", "PERMISSIONUPDATE"]
  })

  schema.mutationType({
    definition(t) {
      t.field('updatePermissions', {
        type: "User",
        nullable: false,
        args: {
          permissions2: schema.arg({ type: "Permission2" }),
          userId: schema.idArg({ nullable: false }),
        },
        resolve: async (_, args, ctx) => {
          return some value
        }
      })
    }
  })
```

which gives me:

```graphql
type Mutation {
  updatePermissions(permissions2: Permission2, userId: ID!): User
}
```

but my original type definition (Prisma 1), which I'm migrating from is:

```graphql
type Mutation {
  updatePermissions(permissions2: [Permission2], userId: ID!): User
}
```

**Reply by ryands17** [reply]
@TheoMer For the above you could write:
```ts
args: {
  permissions2: schema.arg({ type: "Permission2", list: true }),
  userId: schema.idArg({ nullable: false }),
},
```

**Reply by TheoMer** [reply]
@ryands17 Fantastic. Many thanks.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
